lib/utils.ts

```diff
@@ -59,10 +72,21 @@ export function generateBuildTasks(
 				img.image.dockerfile != null
 					? { dockerfilePath: img.image.dockerfile }
 					: {},
-				// It's possible to specify an image name as well as
-				// a build, but that doesn't make sense in a balena
-				// ecosystem, so we remove it
-				_.omit(img.image, 'image'),
+				// Pass through args, context, labels, tag.
+				imageProps,
+				// Pass through build options from composition
+				// translating to dockerode ImageBuildOptions properties
+				{
+					dockerOpts: _.merge(
```

Comment:
To confirm, the purpose of these changes to the `generateBuildTasks` function is to add these properties to `dockerOpts`, right? Is this function the best place to do this though? Maybe it is, but I would have thought that the `runBuildTask` function, where `dockerOpts` is already being modified, would be a better place, because then most handling of `dockerOpts` is done in the same place:
* https://github.com/balena-io-modules/balena-multibuild/blob/v4.12.2/lib/build.ts#L138-L144
* https://github.com/balena-io-modules/balena-multibuild/blob/v4.12.2/lib/build.ts#L164-L188

I think that changing `dockerOpts` in `runBuildTask` has fewer implications on existing logic in `balena-cli` and `balena-builder` (logic that makes assumptions about the "contents" of `dockerOpts` after a call to `fromImageDescriptors`), because `runBuildTask` makes those changes right before sending the Docker API request to the Docker engine.

Reply:
tbh I'm not sure... I don't see how to get the `ImageDescriptor` from `runBuildTask`, since that operates solely on `BuildTask`, so I need to put those options there. I guess I could create new fields, maybe `target` and `buildOpts`. they seem redundant with `dockerOpts` also there though...

Reply:
also, we would then have to reconcile the situation where somebody might pass `network_mode: host` via `BuilTask.dockerOpts` while another value is present in the Composition

Reply:
Fair enough, somehow I thought that `ImageDescriptor` was referenced from a `BuildTask`, but it is not. Sorry for the noise!

Reply:
@pdcastro do you know if we need to worry about this, at least for the cli


> think that changing dockerOpts in runBuildTask has fewer implications on existing logic in balena-cli and balena-builder (logic that makes assumptions about the "contents" of dockerOpts after a call to fromImageDescriptors),

cc @toochevere for the builder

Reply:
Yes, meaning simply that we should check the source code of `balena-cli` and `balena-builder` for how they use `resin-multibuild` and in particular how they set `dockerOpts`. This should be done alongside testing this PR's integration with `balena-cli` and `balena-builder`, before this PR is merged, by creating draft PRs against `balena-cli` and `balena-builder` and using the draft `resin-multibuild` npm package version that balena CI created for this PR, which can be seen in the 'versions' tab at https://www.npmjs.com/package/resin-multibuild

E.g. in the `package.json` of `balena-cli` and `balena-builder`, we would replace:
```json
"resin-multibuild": "^4.12.2",
```
with:
```json
"resin-multibuild": "4.13.0-rgz-pass-build-options-18af5273d0396ada971a8ca8b2b9303781b71ef8",
```
And then check that the automated tests pass for `balena-cli` and `balena-builder`, and do some manual testing of the new features that are made available by this PR.
After we confirm that all CLI and builder tests pass, and the new features work as expected, this PR can be merged and then the `balena-cli` and `balena-builder` PRs can be updated with the final, non-draft version of `resin-multibuild`.